primitives/core/src/crypto.rs

```diff
@@ -704,6 +704,21 @@ impl sp_std::str::FromStr for AccountId32 {
 	}
 }
 
+/// Public key of node which can be used for network connection
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Encode, Decode, crate::RuntimeDebug, PassByCodec)]
+pub enum NodePublicKey {
+	/// An Ed25519 public key
+	Ed25519(ed25519::Public),
+}
+
+impl From<[u8; 32]> for NodePublicKey {
```

Comment:
I feel that `enum` + this `From` is a bit not future proof. If we ever add other variants this might be quite confusing to see `[u8; 32]` being assumed to be `ed25519`. Maybe let's create the enum explicitly?